### ISSUE TITLE
Remove autoselect from autocomplete function

### ISSUE THREAD

**Issue by tfarkas86** [dbeaver/dbeaver#4608]
Autocomplete automatically highlights the first match, and Return will select it to change the text. This is frustrating, because anything typed at the end of a line requires ESC --> RTN to make a new line without changing text to the highlighted autocomplete suggestion. Is there a way to turn off this specific functionality without removing autocomplete entirely? It would be best if TAB were required to enter the autocomplete box, and RTN only then selected autocomplete suggestions.

**Comment by tfarkas86** [comment]
Thanks. I think having Activate On Typing turned off is better than having it turned on. But for the record, (and maybe I'm alone here), I would love if the auto-completion drop-down box appears on typing, but the default is that nothing is selected, so that a Return does not automatically insert the first item in the box. It looks like my preferred "Tab-to-Select" option is almost implemented: if you press Tab, the first item is then highlighted, but it appears to be implicitly selected as soon as the drop-down pops up, so that Return leads to an insert.

**Comment by brent-kelly-vt** [comment]
serge, thanks for the quick-fix solve

tfarkas86, I'm right there with you. "Tab-to-Select" or even "first option always [no-selection]" would be great. The auto-completion is a GREAT feature that helps visibility into not just what's *wanted*, but what's *possible*. Super helpful, especially when working with new data sources.

However, WAY more often than not, it's just in the way, and requires hitting the escape key. 

Primary Example:  Every time I alias a table -- e.g. 'FROM clients AS c' -- when I type the final "c", I want to hit <enter> and move to a new line. Instead, because the autocomplete proposes some table field -- e.g. "client_id" -- I have to hit <esc> then <enter>. 

"<Tab>(or <down-arrow>)-to-enter-the-select menu" or something similiar would be a great option to have.

**Comment by recfish** [comment]
Thanks. I disabled as well. Autocomplete can be extremely useful when implemented correctly, but it does more harm than good here. I think the issue is that it usually completes to the wrong word. For some reason "as" autocompletes to "alembic" for me, which is a table that I never access. I don't even think "as" is an option in the autocomplete list, which leads it to take a best guess even though the letters I actually type contradict its best guess. In addition, if it happens to complete to the right word, it will usually misspell the word. if I select "table_name" from the dropdown, it will give me "table_nameame" carrying over some of my already typed letters. 

In summary to fix the autocorrect feature:
- make sure all possible options appear in the list, so it doesn't incorrectly attempt to autocorrect
- if the word I am spelling no longer matches anything in the list, make sure that pressing enter doesn't choose the best guess
-if I do choose a word from the list, make sure to clean up any trailing letters automatically

lastly, it would be ideal if autocomplete recognized local variables, though I assume this is a more difficult task.

**Comment by uslss** [comment]
there is no updates on the ticket more than a month. Ticket closed.
If you have any ideas that could help us to improve autocompletion (or any other function) - feel free to create feature request :)
